tsdb/shard.go

```diff
@@ -283,20 +528,18 @@ func (s *Shard) deleteMeasurement(name string, seriesKeys []string) error {
 			if err := b.Delete([]byte(k)); err != nil {
 				return err
 			}
-			if err := tx.DeleteBucket([]byte(k)); err != nil {
+			if err := tx.DeleteBucket([]byte(k)); err != nil && err != bolt.ErrBucketNotFound {
 				return err
 			}
+			delete(s.cache, k)
 		}
 
 		return nil
 	}); err != nil {
-		_ = s.Close()
 		return err
 	}
 
 	// Remove entry from shard index.
-	s.mu.Lock()
```

Comment:
Bucket deletion is fast but more importantly it keeps the in-memory stuff and the Bolt DB data in sync.